src/renderer/app-update-warning/app-update-warning.injectable.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import ipcRendererInjectable from "../app-paths/get-value-from-registered-channel/ipc-renderer/ipc-renderer.injectable";
+import sessionStorageInjectable from "../utils/session-storage.injectable";
+import { AppUpdateWarning } from "./app-update-warning";
+
+const appUpdateWarningInjectable = getInjectable({
+  id: "app-update-warning",
+
+  instantiate: (di) => {
+    AppUpdateWarning.resetInstance();
+    
+    return AppUpdateWarning.createInstance({
+      ipcRenderer: di.inject(ipcRendererInjectable),
```

Comment:
Why `ipcRenderer` needs to injectable?

Reply:
So that in every unit test it can be a fresh mock. Though it is probably better if this was instead a `SyncBox` or some `IComputedValue` derived from the `SyncBox`'s that Janne and Mikko introduced in #5433

Reply:
Unit tests takes too much.. We end-up that way having injectable for everything what is "better for tests"..

Reply:
We're moving to direction where `ipc` is going to be technical detail from the perspective of operational code and unit tests. We'll be creating easy-to-use tools on top of that (e.g. `SyncBox` that @Nokel81 mentioned) which make sure that you don't need to know anything about the technical implementation. 

This implementation shouldn't use `ipcRenderer` at all. There is better alternatives for the job. Currently we have low-level implementations: `messageToChannel`, `requestFromChannel` and higher level implementation: `SyncBox` which can be utilized.

Reply:
I don't suggest to use `ipcRenderer` directly from import.. But single abstraction in module/object/function would be enough imho.

Reply:
> Unit tests takes too much.. We end-up that way having injectable for everything what is "better for tests"..

Remember, unit testing alone is a lot, but it is by far not the only motivation to organise code like this (i.e. Dependency Inversion Principle). We also get stuff like simplicity, modularity, high cohesion, low coupling, composability, maintainability, reusability, clean abstractions, controllable side-effects, deterministic implementations, clean water, sandy beaches and margaritas.

Reply:
I don't think simple abstraction-layer/wrapper is re-inventing the wheel, there is nothing to invent, it does the job and incapsulates "native api" where we hide it and keep simple and in single place / source-of-truth.
That way  at some point you can replace for example `IPC` to `REST` in single module/place. 
If framework/tools becomes too opinionated they becomes burden.

> With correct tools, you can just inject the function and call it

With correct implementation you just import module and it just works™

Reply:
> I don't think simple abstraction-layer/wrapper is re-inventing the wheel, there is nothing to invent, it does the job and incapsulates "native api" where we hide it and keep simple and in single place / source-of-truth.
That way at some point you can replace for example IPC to REST in single module/place without a hassle.
If framework/tools becomes too opinionated they becomes burden.

That's what we are doing here. Instead of introducing single abstraction layer for a tool, we are introducing one abstraction layer for purpose and using a tool to implement that abstraction. That means that we are able to change our mind which tool we are using for what purpose. One purpose at the time.

> With correct implementation you just import module and it just works™

Well, that would lead to place where you need to specify the dependencies when called. If that's not the case, we couldn't depend from the interfaces and would have to depend from the implementations. Therefore we wouldn't be able to replace one implementation with something else which just fulfills the required interface.

One common used example is how you access database. You shouldn't depend from e.g. DynamoDB, you should depend from interface that gets implemented with DynamoDB stuff. (or file system, in-memory, whatever.)